### ISSUE TITLE
Implementing a `Protocol` field with `property` is not prohibited

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#2071]
I think implementing a `Protocol` mutable field with `property` must be generally prohibited because a simple `property` is not mutable.
The code below shows a protocol `P` and its implementers `Foo` and `Bar`:

```python
class P(Protocol):
    v: int

@dataclass
class Foo(P):
    v: int # It is ok, of course

class Bar(P):
    @property
    def v(self) -> int: # This must not be allowed (mypy denies it)
        return 1

def f(x: P):
    x.v = 0

foo = Foo(1)
f(foo)

bar = Bar()
f(bar) # Runtime Error: AttributeError: can't set attribute
```

Pyright accepts this code, while Mypy denies it.

**Comment by erictraut** [comment]
Pyright will report this as an error if you enable the `reportIncompatibleVariableOverride` diagnostic check.
